addons/mail/static/src/chatter/web/mail_composer_form.js

```diff
@@ -27,6 +30,24 @@ export class MailComposerFormRenderer extends formView.Renderer {
             this.root,
             this.props.record.resId
         ]);
+
+        // Add the file dropzone on the mail composer.
+        this.uploadFiles = useFileUploader();
+        this.operations = useX2ManyCrud(() => {
+            return this.props.record.data["attachment_ids"];
+        }, true);
+        useDropzone(this.root, async event => {
+            const { files } = event.dataTransfer;
+            const attachments = await this.uploadFiles("/web/binary/upload_attachment", {
+                csrf_token: odoo.csrf_token,
+                ufile: [...files],
+                model: this.props.record.resModel,
+                id: this.props.record.resId || 0,
```

Comment:
I added the `|| 0` because the route `/web/binary/upload_attachment` raise an exception when casting the string `false` to an integer.

```
Fail to upload attachment my_document.pdf
Traceback (most recent call last):
  File "/home/odoo/src/odoo/addons/web/controllers/binary.py", line 234, in upload_attachment
    'res_id': int(id)
```

For that reason, I assume that the mail composer does not have any `resId` when opening it.

When I open the chatter of an article in Knowledge, open the full mail composer, drop a file, I see that an attachment is created in `Settings / Technical / Database Structure / Attachments`. The attachment is linked to the `mail.compose.message` model and does not have any `resId`. When I send the message, I see that the attachment gets linked to the corresponding article.

So, it does not seem to be a problem to have no `resId`. I indeed get the message with the attachments :smiley: